File: src/connectivity/network/netstack3/core/src/transport/tcp/socket/icmp.rs
// Copyright 2022 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

//! Defines how TCP sockets interacts with incoming ICMP messages.

use log::error;
use net_types::SpecifiedAddr;

use crate::{
    ip::{icmp::IcmpIpExt, IpDeviceIdContext, IpTransportContext},
    transport::tcp::socket::TcpIpTransportContext,
};

impl<I: IcmpIpExt, C, SC: IpDeviceIdContext<I> + ?Sized> IpTransportContext<I, C, SC>
    for TcpIpTransportContext
{
    fn receive_icmp_error(
        _sync_ctx: &mut SC,
        _ctx: &mut C,
        _device: &SC::DeviceId,
        _original_src_ip: Option<SpecifiedAddr<I::Addr>>,
        _original_dst_ip: SpecifiedAddr<I::Addr>,
        _original_body: &[u8],
        err: I::ErrorCode,
    ) {
        // TODO(https://fxbug.dev/101806): Process incoming ICMP error message.
        error!(
            "TcpIpTransportContext::receive_icmp_error: Received ICMP error message ({:?}) for TCP",
            err
        );
    }
}

File: src/connectivity/network/testing/conformance/expectation/icmp.go
// Copyright 2022 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package expectation

import "go.fuchsia.dev/fuchsia/src/connectivity/network/testing/conformance/expectation/outcome"

var icmpExpectations map[AnvlCaseNumber]outcome.Outcome = map[AnvlCaseNumber]outcome.Outcome{
	{1, 1}:  Pass,
	{1, 5}:  Pass,
	{2, 1}:  Pass,
	{2, 2}:  Pass,
	{2, 3}:  Pass,
	{2, 4}:  Pass,
	{2, 5}:  Pass,
	{3, 1}:  Pass,
	{4, 2}:  Pass,
	{4, 3}:  Fail,
	{4, 4}:  Pass,
	{5, 1}:  Pass,
	{5, 2}:  Pass,
	{5, 3}:  Pass,
	{8, 1}:  Pass,
	{8, 2}:  Pass,
	{8, 3}:  Pass,
	{10, 1}: Pass,
}
